src/simplification/Simplifier.ts

```diff
@@ -44,9 +44,13 @@ export class Simplifier {
       this.simplifyModel(model, schema);
     }
     this.ensureModelsAreSplit(model);
+    //Ensure current model is not part of the iterated list since we could have circular schemas
+    if (this.iteratedModels[`${model.$id}`] !== undefined) {
+      delete this.iteratedModels[`${model.$id}`];
+    }
```

Comment:
Can I trust you? With next PRs, for me logic is more and more magic 😄

Reply:
Never trust anyone 😈 

The problem is at line 51 we create a copy of all the values of the `iteratedModels` and if we have a circular schema that we would return the same model twice `return [model1, ...[model2, model1]];`

Unfortunately, I don't see any other way around this based on the nature of JSON Schema 😟

Reply:
I understand the logic, but it seems more and more complex and I see that it (probably) will cause a lot of errors in the future. We must to think about another way to handle scheme processing.

Reply:
The only other way to do this is by not using recursion and that would be a rewrite from the ground up and basically reengineered 😄

Reply:
Yeah, by `We must to think about another way to handle scheme processing` I mean reengineered. What we have learned is ours 😄